src/main/haskell/kore/src/Kore/Step/Function/Matcher.hs

```diff
@@ -149,6 +151,9 @@ unificationWithAppMatchOnTop tools substitutionSimplifier first second
                         tools
                         substitutionSimplifier
                         (zip firstChildren secondChildren)
+                else if symbolOrAliasConstructor firstHead
```

Comment:
Because we are growing a longer list of cases here, I think the body of this `case` branch would be better written as a sequence of guards.